app/scripts/controllers/swaps.js

```diff
@@ -184,11 +190,18 @@ export default class SwapsController {
       }
     }
 
+    // If a newer call has been made, don't update state with old information
+    // Prevents timing conflicts between fetches
+    if (this.indexOfNewestCallInFlight !== indexOfCurrentCall) {
+      throw new Error('Swaps fetch order conflict')
```

Comment:
Looks good, now we just need to handle this here https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/ducks/swaps/swaps.js#L402

So that we don't get in an error state.